Update disk encryption enforcement details for Windows (#43915)
fleetdm/fleet: articles/enforce-disk-encryption.md

For macOS hosts that automatically enroll, end users are forced to enable disk encryption during Setup Assistant. For hosts that manually enroll, end users are forced to enable disk encryption the next time they log out and log back in. For both enroll methods, end users can't defer. 

> For Windows, currently, disk encryption is enforced on the C: volume (default system/OS drive) only on hosts with a [TPM chip](https://support.microsoft.com/en-us/topic/what-s-a-trusted-platform-module-tpm-705f241d-025d-4470-80c5-4feeb24fa1ee). BitLocker PIN can only be enforced on hosts that do not have the C: volume already encrypted.

Reviewer: ```suggestion
On Windows, disk encryption is currently enforced only on the C: volume (the default system/OS drive) for hosts with a [TPM chip](https://support.microsoft.com/en-us/topic/what-s-a-trusted-platform-module-tpm-705f241d-025d-4470-80c5-4feeb24fa1ee). A BitLocker PIN can only be enforced on hosts that haven't yet been set up, since their C: volume has not yet been encrypted.
```

Reviewer: Flipped the language to "haven't yet been setup". I think that's clearer to IT admins.

@getvictor did we test this? Is the BitLocker PIN enforced on new Windows hosts?

Author: @getvictor ^ ?

Author: We have a research story to understand this. Until then, I will convert this PR draft.

https://github.com/fleetdm/fleet/issues/44211

> For Linux, encryption requires end user interaction.

Reviewer: ```suggestion
For Linux, encryption requires end user interaction. [Learn more](#enforce-disk-encryption-on-linux).
```
